pkg/providers/aws.go

```diff
@@ -28,19 +29,30 @@ var _ = Describe("[Feature:Machines] [AWS] MetadataServiceOptions", func() {
 
 	toDelete := make([]*machinev1.MachineSet, 0, 3)
 
+	gatherer, err := framework.NewGatherer()
+	Expect(err).ToNot(HaveOccurred())
+
 	BeforeEach(func() {
 		platform, err := framework.GetPlatform(client)
 		Expect(err).ToNot(HaveOccurred())
 		if platform != configv1.AWSPlatformType {
 			Skip(fmt.Sprintf("skipping AWS specific tests on %s", platform))
 		}
+
+		// Setup new 'since time' for data gatherer for minimize capturing unrelevant to particular test pod logs
+		gatherer = gatherer.WithSinceTime(time.Now())
 	})
 
 	AfterEach(func() {
 		Expect(framework.DeleteMachineSets(client, toDelete...)).To(Succeed())
 		toDelete = make([]*machinev1.MachineSet, 0, 3)
 
 		framework.WaitForMachineSetsDeleted(client, toDelete...)
+
+		testDescription := CurrentGinkgoTestDescription()
+		if testDescription.Failed == true {
+			Expect(gatherer.WithTestDescription(testDescription).GatherAll()).To(Succeed())
```

Comment:
Will the gatherer output go into a folder per test?

Reply:
Yup.